src/Mvc/test/WebSites/BasicWebSite/Controllers/ContactApiController.cs

```diff
@@ -83,6 +83,9 @@ public ActionResult<Contact> ActionWithInferredEmptyPrefix([FromQuery] Contact c
         return foo;
     }
 
+    [HttpGet("[action]/{id}")]
+    public ActionResult<Contact> ActionWithInferredFromServicesParameter(int id, ContactsRepository repository) => repository.GetContact(id) ?? new Contact() { ContactId = id };
```

Comment:
We don't enforce line size limits, but it's preferable to split it in a way to not have a horizontal scroll bar when reviewing code on GitHub:


```suggestion
    public ActionResult<Contact> ActionWithInferredFromServicesParameter(int id, ContactsRepository repository) 
        => repository.GetContact(id) ?? new Contact() { ContactId = id };
```